Objects/unicodeobject.c

```diff
@@ -290,12 +290,6 @@ init_interned_dict(PyInterpreterState *interp)
 static void
 clear_interned_dict(PyInterpreterState *interp)
 {
-    PyObject *interned = get_interned_dict(interp);
-    if (interned != NULL) {
-        PyDict_Clear(interned);
-        Py_DECREF(interned);
-        _Py_INTERP_CACHED_OBJECT(interp, interned_strings) = NULL;
-    }
```

Comment:
If I understand the crash report correctly, the problem is that the interned dict is cleared for sub-interpreters when it is also shared between the main interpreter and sub-interpreters. This change means that PyDict_Clear() is never called, not even for the main interpreter. That seems like it will just leak everything in the dict.

I imagine, rather, that you'd put this code inside the block below that is guarded by the _Py_IsMainInterpreter() checl.

Reply:
Thanks for the review! Sorry, I put this PR too earlier, after some research I found that I was wrong, and the crash is caused by other place.

For now I can ensure that this line https://github.com/python/cpython/blob/34f5ae69fe9ab0f5b23311d5c396d0cbb5902913/Python/import.c#L1973 caused the crash. It will intern the file name for the `_tkinter.so`, and after the interned dict finalization, call free on it will crash.

I guess maybe this unicode object is created in another interpreter, and intern it in the main interpreter will cause the error. But I'm not familiar with these code, and still working on it.